gluten-core/src/main/scala/io/glutenproject/expression/QuaternaryExpressionTransformer.scala

```diff
@@ -24,6 +24,41 @@ import io.glutenproject.substrait.expression.{ExpressionBuilder, ExpressionNode}
 
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.RegExpReplace
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.GlutenConfig
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.types.IntegerType
+import io.glutenproject.substrait.expression.IntLiteralNode
+
+class RegExpReplaceTransformer(subject: ExpressionTransformer, regexp: ExpressionTransformer,
+  rep: ExpressionTransformer, pos: ExpressionTransformer, original: RegExpReplace)
+  extends ExpressionTransformer with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(GlutenConfig.GLUTEN_VELOX_BACKEND)) {
+      QuaternaryExpressionTransformer(ExpressionMappings.REGEXP_REPLACE,
+        subject, regexp, rep, pos, original).doTransform(args)
+    }
+
+
+    val posNode = pos.doTransform(args)
+    if (!posNode.isInstanceOf[IntLiteralNode] ||
+      posNode.asInstanceOf[IntLiteralNode].getValue() != 1) {
+      throw new UnsupportedOperationException(s"${original} not supported yet.")
+    }
+
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionName = ConverterUtils.makeFuncName(ExpressionMappings.REGEXP_REPLACE,
```

Comment:
use `substraitExprName: String` to pass substrait expression name, dont use `ExpressionMappings.REGEXP_REPLACE`